### ISSUE TITLE
Random Windows node stops working with "NetworkPlugin cni failed to set up pod ..." error

### ISSUE THREAD

**Issue by ai2admin** [Azure/acs-engine#4331]
**Is this a request for help?**: yes

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): ISSUE

---

**What version of acs-engine?**:  v0.25.3

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes, Windows nodes 1709

**What happened**:

Nodes randomly start failing in a few days of use. New pods stop being created on a particular node, with the error being Failed create pod sandbox: rpc error: code = Unknown desc = NetworkPlugin cni failed to set up pod "my-pod-648b47bcc-hj8tm_t-1ea5c65c-0824-4d63-8f5f-f8e820322e28" network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input

Same deployment succeeds on other nodes.

kubectl get nodes returns node status as "Ready".

Restarting the node doesn't help. Recreating the cluster fixes the issue for a few days, then a random node goes bad again

**What you expected to happen**:

Nodes should be stable

**How to reproduce it** (as minimally and precisely as possible):

Create a cluster with a few Windows nodes, use it with a few pods. One of the nodes will stop creating new pods in a few days.

**Anything else we need to know**:

The node shows no resource stress.

**Comment by zhiweiv** [comment]
Try #4046

**Comment by ai2admin** [comment]
@zhiweiv, there are 4 or 5 different suggestions given at #4046. Which one?

**Comment by zhiweiv** [comment]
Stop-Service Kubelet -Force

<del azure-vnet.json file, if exist>
<del azure-vnet-ipam.json file, if exist>
delete the lock files as well.

Start-Service kubelet

**Comment by zhiweiv** [comment]
I got same issue again today, the workaround worked for me.

**Comment by ai2admin** [comment]
Yes, the fix also worked in my case. The node hadn't crashed before the problem appeared though.

**Comment by daschott** [comment]
Can you please use azure CNI v1.0.16 and confirm whether this issue is resolved?

Also, for node stability issues for WS2019, there is a hotfix that should come out on 01/29 as KB4476976.

**Comment by ai2admin** [comment]
@daschott , thank you David. Do need to redeploy my cluster with "azureCNIVersion": "v1.0.16" value in apimodel.json generated by aks-engine? Any way to do it for an already existing cluster so that there is not much downtime involved?

**Comment by Gothic-X** [comment]
Here is the expanded steps I ran to address this issue

Note: Only do this after you deleted or cordoned any pods running on any of the Windows node

1) Attach a public IP address to the Windows node
    Steps: How to add a public IP address to a VM: https://www.mismosystems.com/2017/12/26/add-static-public-ip-address-to-an-existing-azure-vm/

2) RDP to the Windows node (Note: Credentials in apimodel.json)

3) Run the following steps inside the command window:

    a) Powershell
    b) cd \
    c) Stop-Service Kubelet -Force
    d) Invoke-WebRequest -Uri https://github.com/Azure/azure-container-networking/releases/download/v1.0.16/azure-vnet-cni-windows-amd64-v1.0.16.zip -OutFile c:\azure-vnet-cni-windows-amd64-v1.0.16.zip
    e) Expand-Archive c:\azure-vnet-cni-windows-amd64-v1.0.16.zip
    f) cd c:\azure-vnet-cni-windows-amd64-v1.0.16
    g) copy * c:\k\azurecni\bin\
    h) copy 10-azure.conflist c:\k\azurecni\netconf\
    i) cd c:\k
    j) del azure-vnet*.lock
    k) del azure-vnet*.json
    l) Start-Service Kubelet

5) Terminate the RDP session

6) Detach the public IP address from the Windows node

7) Reboot VM
